docs/synapse/userguides/ug012_storm_ref_datamod.rst

```diff
@@ -0,0 +1,299 @@
+
+
+setprop()
+---------
+
+Sets one or more property values on the specified node(s).
+
+**Operator Syntax:**
+
+.. parsed-literal::
+  *<query>* **setprop( :** *<prop>* **=** *<pval>* **,** ... **)**
+
+**Macro Syntax:**
+
+.. parsed-literal::
+  *<query>* **[ :** *<prop>* **=** *<pval>* ... **]**
+
+**Examples:**
+
+``inet:dns:a = woot.com/1.2.3.4 setprop( :seen:min = "2017-08-01 01:23" , :seen:max = "2017-08-10 04:56" )``
+
+``inet:dns:a = woot.com/1.2.3.4 [ :seen:min = "2017-08-01 01:23" :seen:max = "2017-08-10 04:56" ]``
+
+**Usage Notes:**
+
+* ``setprop()`` operates on the output of a previous Storm query.
+* Secondary properties must be specified by their relative property name. For the form ``foo:bar`` and the property ``baz`` (e.g., ``foo:bar:baz``) the relative property name is specified as ``:baz``.
+* Synapse will set the secondary propert(ies) for all nodes returned by ``<query>`` for which that secondary property is a valid property. Nodes for which that property is not a valid secondary property will be ignored.
+* ``setprop()`` will create and set the property if it does not exist, or overwrite the existing ``<prop>=<pval>`` if it does exist.
+* ``setprop()`` can set or modify any property not explicitly defined as read only (``'ro' : 1``) in the data model. Attempts to modify read only properties will fail silently (e.g., the property value will not be overwritten, but the user will not be notified that the request failed).
+* ``setprop()`` cannot be used to remove (delete) a property entirely.
+
+**Operator Syntax Notes:**
+
+* N/A
+
+**Macro Syntax Notes:**
+
+* Synapse will attempt to set the specified propert(ies) for all previously referenced nodes (e.g., to the left of the ``<prop>=<pval>`` statement) for which that property is valid, **whether those nodes are within or outside of the macro syntax brackets.** See `Special Note on Macro Syntax`_.
+
+addtag()
+--------
+
+Adds one or more tags to the specified node(s).
+
+**Operator Syntax:**
+
+.. parsed-literal::
+  *<query>* **addtag(** *<tag>* [ **,** ... ] **)**
+
+**Macro Syntax:**
+
+.. parsed-literal::
+  *<query>* **[** **#** *<tag>* ... **]**
+
+**Examples:**
+
+``inet:fqdn = woot.com addtag( foo.bar , baz.faz )``
+
+``inet:fqdn = woot.com [ #foo.bar #baz.faz ]``
+
+**Usage Notes:**
+
+* ``addtag()`` operates on the output of a previous Storm query.
+* Synapse will apply the specified tag(s) to all nodes returned by ``<query>``.
+
+**Operator Syntax Notes:**
+
+* N/A
+
+**Macro Syntax Notes:**
+
+* Synapse will set the specified tag(s) for all previously referenced nodes (e.g., to the left of the ``<tag>`` statement) **whether those nodes are within or outside of the macro syntax brackets.** See `Special Note on Macro Syntax`_.
+
+delnode()
+---------
+
+Deletes the specified node(s) from a Cortex.
+
+**Operator Syntax:**
+
+.. parsed-literal::
+  *<query>* **delnode(force=1)**
```

Comment:
force is not required in order to run delnode.  force=1 is an optional arg that actually causes the node deletion to occur.  Otherwise this operator just consumes the current nodes in the query and does nothing with them.

Reply:
this is still showing force=1 as required - do we want to change it to be an optional kwarg as per the syntax spec?

Reply:
fixed - thanks for the catch